mergify_engine/backports.py

```diff
@@ -161,7 +161,7 @@ def backport(pull, branch, installation_token):
             "checking-out-pull-requests-locally/")
 
     return repo.create_pull(
-        title="Automatic backport of pull request #%d" % pull.g_pull.number,
+        title="[bp] %s" % pull.g_pull.title,
```

Comment:
This solution is not 100% correct.

GitHub automatically trims the text from the title of the PR in order to append `(#pr-num)`. Since this change appends `[bp]` (to signal it's a `backport`) it is possible the resulting title overflows the [50 characters](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html). I'll gladly work on a function that rewrites the title to avoid overflows.

Reply:
What about just using the original title?
The target branch is different anyway, so that might not be a problem.

You could also use a template system in the action where `{{title}}` would be replaced by the original title. So everyone can use their own backport title.

Reply:
Great @ignasi35! 

@jd I like the idea of templating. 

I also find it useful to see somewhere that this is a backport of some other PR, but that can be solved by a template. Each project can choose how they want the BP message.

Reply:
I completely agree that templating would be the long term goal but I think it's a bit out of my reach ATM. I'd like to focus on the baby step first.

I've reviewed the behavior of `title="Automatic backport of pull request #%d" % pull.g_pull.number` by observing an actual [example](https://github.com/lagom/lagom/pull/1891) and noticed `pull.g_pull.number` is the number of the original PR. I am assuming, then, that `pull.g_pull.title` is the title of the original PR. If that's the case, `pull.g_pull.title` will _not_ contain a `(#num)` at the end, so my comment above is wrong.

Taking [this PR](https://github.com/lagom/lagom/pull/1890) as an example:

 * the original PR number is `#1890`
 * the original PR title is `Upgrades to Play 2.7.2`

My intent is for mergify to produce the backport PR with the title `Upgrades to Play 2.7.2 (bp #1890)`. That is `"{} (bp #{})".format(pull.g_pull.title, pull.g_pull.number)` (using [`.format()` notation](https://pyformat.info/)).

@jd It'd be great if you could provide some guidance for me to write and run some tests locally. I'm running a bit blind. Again, first-timer on python-world 😅.

Reply:
You can run the test locally by using `tox -e py37`, assuming you have Python 3.7 installed.

Since your change did not break any test, I don't think we have a test checking the title of backported PR. That sounds not a big issue for me, and adding more tests is complicated  — the tests have to interact with `github.com` which needs special tests accounts).

So I'm good with that for now. I'll @sileht comment.